code/__HELPERS/cidtest.dm

```diff
@@ -0,0 +1,134 @@
+
+
+var/global/compid_file
+
+
+/datum/compid_info
+	var/compid = ""
+	var/last_seen = 0 //The time this was last spotted
+	var/last_ckey = "" //The latest key associated with this ID
+	var/times_seen = 0
+
+/proc/initialize_compid_savefile()
+	if(!compid_file)
+		compid_file = new /savefile("data/compid_file.sav")
+
+	return compid_file
+
+/proc/load_compids(var/ckey)
+	var/savefile/SF = initialize_compid_savefile()
+
+	var/path = "/[copytext(ckey, 1, 2)]/[ckey]"
+	var/list/datum/compid_info/cid_list = list()
+
+	SF.cd = path
+	while (!SF.eof)
+		var/datum/compid_info/CI
+		SF >> CI
+		cid_list += CI
+
+	return cid_list
+
+/proc/save_compids(var/ckey, var/list/datum/compid_info/cid_list)
+	var/savefile/SF = initialize_compid_savefile()
+	var/path = "/[copytext(ckey, 1, 2)]/[ckey]"
+
+	SF.cd = path
+	SF.eof = -1
+	for(var/datum/compid_info/CI in cid_list)
+		SF << CI
+
+/proc/check_compid_list(var/client/C)
+	C.compid_info_list = load_compids(C.ckey)
+
+	var/append_CID = 1
+
+	for(var/datum/compid_info/CI in C.compid_info_list)
```

Comment:
actually istypeless loop is also wrong goof, there should be NO loop, it should just be assoc.